### PR TITLE
Add variable for publishing egress IP addresses

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR adds a new Terraform variable (`publish_egress_ip_addresses`) that specifies whether EC2 instances in the operations subnet should be tagged to indicate that their public IP addresses may be published.

## 💭 Motivation and context ##

This is useful for deconfliction purposes and it allows us to control publishing of IP addresses on a per-assessment (per-Terraform workspace) basis, which is a requirement of https://github.com/cisagov/cool-system-internal/issues/125.

IP address publishing will be handled via the code in [`cisagov/publish-egress-ip-lambda`](https://github.com/cisagov/publish-egress-ip-lambda) and [`cisagov/publish-egress-ip-terraform`](https://github.com/cisagov/publish-egress-ip-terraform).

## 🧪 Testing ##

I tested this by running a `terraform plan` for an environment with:
* `publish_egress_ip_addresses = true`
* No value set for `publish_egress_ip_addresses`

In the first case, I confirmed that the plan wanted to set `"Publish Egress" = "true"` for all of the correct instances/EIPs.
In the second case, I confirmed that the plan wanted to set `"Publish Egress" = "false"` for all of the correct instances/EIPs.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Post-merge checklist ##

- [x] Coordinate with the COOL Operations team to ensure that this new variable is used correctly.
- [x] Update all Staging and Production tfvars files to include `publish_egress_ip_addresses`

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]

**Review by dv4harr10 — APPROVED** [review]

File: debiandesktop_ec2.tf
Comment (mcdonnnj) [inline debiandesktop_ec2.tf:60]: Will this value massage to the `True` or `False` that was in use before? I'm concerned this will interpolate to a string version of `true` or `false` which doesn't quite match up.

**Review by mcdonnnj — COMMENTED** [review]
This seems pretty solid. In addition to my one feedback item I would like to raise maybe making an issue to expand functionality to allow control over which instances of a given type have their IP information published e.g. I want `Windows0` to be private but `Windows1` to be published.

File: debiandesktop_ec2.tf
Comment (dav3r) [reply debiandesktop_ec2.tf:60]: You are good!  After this PR is merged, I was planning to modify the lambda code [here](https://github.com/cisagov/publish-egress-ip-lambda/blob/bfb38499afdb29ac622f687a364088e8bf19e119/src/lambda_handler.py#L97-L110) to account for this change from `True` to `true`.

I already had it written, but haven't committed it yet:

Original:
```python
        if tags.get(publish_egress_tag_name, str(False)) != str(True):
```
New:
```python
        if tags.get(publish_egress_tag_name, "false").lower() != "true":
```

File: pentestportal_ec2.tf
Comment (mcdonnnj) [inline pentestportal_ec2.tf]: Based on https://github.com/cisagov/cool-assessment-terraform/pull/240#discussion_r1600570216 I think this should be changed.

```suggestion
    "Publish Egress" = "false"
```

Comment (dav3r) [reply pentestportal_ec2.tf]: Anything other than `true` won't actually matter, but I still agree with you. 😄

**Comment by dav3r** [comment]
> This seems pretty solid. In addition to my one feedback item I would like to raise maybe making an issue to expand functionality to allow control over which instances of a given type have their IP information published e.g. I want `Windows0` to be private but `Windows1` to be published.

@mcdonnnj That's a good idea and we should keep it in mind, but until someone comes to us with that as a requirement, I don't think we need to worry too much about it.

Do I have your approval on this PR?

**Comment by mcdonnnj** [comment]
> > This seems pretty solid. In addition to my one feedback item I would like to raise maybe making an issue to expand functionality to allow control over which instances of a given type have their IP information published e.g. I want `Windows0` to be private but `Windows1` to be published.
> 
> @mcdonnnj That's a good idea and we should keep it in mind, but until someone comes to us with that as a requirement, I don't think we need to worry too much about it.
> 
> Do I have your approval on this PR?

Yeah I meant it as a "feature idea" issue just to document the thought.

**Review by mcdonnnj — APPROVED** [review]
LGTM ✔

**Comment by dav3r** [comment]
> > > This seems pretty solid. In addition to my one feedback item I would like to raise maybe making an issue to expand functionality to allow control over which instances of a given type have their IP information published e.g. I want `Windows0` to be private but `Windows1` to be published.
> > 
> > 
> > @mcdonnnj That's a good idea and we should keep it in mind, but until someone comes to us with that as a requirement, I don't think we need to worry too much about it.
> > Do I have your approval on this PR?
> 
> Yeah I meant it as a "feature idea" issue just to document the thought.

Understood - I put this idea in https://github.com/cisagov/cool-assessment-terraform/issues/242.  Feel free to make any edits as you see fit.

**Pull request merged**
